### DISCUSSION TITLE
How do you limit the number of rows

### DISCUSSION THREAD

**Discussion by aszenz** [perspective-dev/perspective#2556] [Q&A]
How can I get top 5 and bottom 5 results in the grid?, is there a way to limit the number of rows, this is quite handy especially for pie charts

**Comment by aszenz** [comment]
@texodus Is there a limit mechanism for rows in the underlying query engine that could solve this

**Comment by Cloud500** [comment]
I have found a solution.

You can edit the properties of a plugin within a perspective-viewer element and have a max_cells property there.

Example:
``` JavaScript
const viewer = document.getElementById("my-viewer");
const xBarPlugin = viewer.getPlugin("X Bar");
plugin.max_cells = 15; // if necessary +1 some plugins show one cell less than specified
```

Result:
![2024-05-05 07_19_26-Window](https://github.com/finos/perspective/assets/10844300/8317560c-f571-4c81-9721-a1578d9e0d6c)
Ignore the clunky, cheap-looking Wordcloud >.<

Now the only thing left to do is to find a nice way to turn off the "Render x% of Points." warning ^^
My temporary solution is to simply hide the element:
``` JavaScript
viewer.shadowRoot.querySelector('div#plugin_information--size').style.display = "none";
```

However, I can't say yet if this works for all plugins, I have only used/tested it for "X Bar".

I hope this can help you

**Reply by aszenz** [reply]
Cool, when creating a perspective table one can pass a limit like this:

```js
const _table = worker.table(data.buffer, { limit: 10 });
```

But I would have liked is to pass the limit to a perspective view just like filter and order by.

Your solution does indeed work in this case, even though it is a workaround.
